services/workflows-service/src/document/document.service.ts

```
@@ -683,21 +717,31 @@ export class DocumentService {
         return context;
       }
 
-      return addRequestedDocumentToEntityDocuments(
-        context,
-        document.type as 'business' | 'ubo' | 'director',
-        uiDefinition,
-        {
-          id: createdDocument.id,
-          status: DocumentStatus.requested,
-          decision: null,
-          version: createdDocument.version.toString(),
-          type: createdDocument.type,
-          category: createdDocument.category,
-          issuingCountry: createdDocument.issuingCountry,
-          issuingVersion: createdDocument.issuingVersion,
-        },
-      );
+      const documentToInsert = {
+        id: createdDocument.id,
+        status: DocumentStatus.requested,
+        decision: null,
+        version: createdDocument.version.toString(),
+        type: createdDocument.type,
+        category: createdDocument.category,
+        issuingCountry: createdDocument.issuingCountry,
+        issuingVersion: createdDocument.issuingVersion,
+        entityId: createdDocument.entityId as string | undefined,
+      };
+
+      return document.entityType === 'business'
+        ? addRequestedDocumentToBusinessEntityDocuments(
+            context,
+            document.entityType as 'business' | 'ubo' | 'director',
+            uiDefinition,
+            documentToInsert,
+          )
+        : addRequestedDocumentToIndividualDocuments(
+            context,
+            document.entityType as 'ubo' | 'director',
+            uiDefinition,
+            documentToInsert,
+          );
```

Let's throw an error if the entity type is unsupported. There are other places in the system that use a ternary we're trying to remove over time.